### PR TITLE
Issue #638: Fix all our tests

### PR DESCRIPTION
**Pull request by justafish**
- Replace sserbin/twig-linter with vincentlanglet/twig-cs-fixer/ - sserbin/twig-linter depends on an old version of symfony/console and core is adopting this new package at some point https://www.drupal.org/project/drupal/issues/3284817 - it also has a much better extension mechanism which will help if something like #615 comes up again
- Update Pantheon settings for Drupal 11
- Update PHPUnit configurations and tests for Drupal 11
- Fix mysterious yarn issues with DDEV

### CODE REVIEW

File: .github/workflows/TestPHPUnit.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
         uses: GuillaumeFalourd/assert-command-line-output@v2.3
         with:
           command_line: xmllint --xpath 'string(//testsuites/testsuite/@tests)' test_result/phpunit-functional.xml
-          contains: 18
+          contains: 6
```
Comment (justafish) [inline .github/workflows/TestPHPUnit.yml:81]: These were previously also running the unit tests apparently

File: .github/workflows/TestPHPUnit.yml [hunk 2]
```diff
@@ -147,7 +147,7 @@ jobs:
         uses: GuillaumeFalourd/assert-command-line-output@v2
         with:
           command_line: xmllint --xpath 'string(//testsuites/testsuite/@tests)' test_result/phpunit-functional.xml
-          contains: 21
+          contains: 9
```
Comment (justafish) [inline .github/workflows/TestPHPUnit.yml:150]: These were previously also running the unit tests apparently

**Comment by deviantintegral** [comment]
You've buried the lede here with "Drupal 11 compatibility"!

File: .github/workflows/TestMetapackages.yml [hunk 3]
```diff
@@ -271,10 +273,12 @@ jobs:
...
         run: |
-          ddev yarn set version classic
-          ddev yarn init -y
-          ddev yarn add focus-trap@^6.7.3
-          ddev yarn add file:./drainpipe/metapackages/javascript --dev
+          ddev exec corepack disable
+          ddev exec npm i -g yarn@1.22.1
+          ddev exec 'ln -s $(npm root -g)/yarn/bin/yarn /usr/local/bin/yarn'
+          ddev exec yarn init -y
+          ddev exec yarn add focus-trap@^6.7.3
```
Comment (deviantintegral) [inline .github/workflows/TestMetapackages.yml:280]: what's this package for? Is it just an easy way to do a test?

File: .github/workflows/test-production-build.yml [hunk 4]
```diff
@@ -12,7 +12,11 @@ jobs:
     strategy:
       matrix:
         php-version: [8.1, 8.2, 8.3]
-
+        include:
+          - php-version: 8.1
+            drupal-version: ":^10"
```
Comment (deviantintegral) [inline .github/workflows/test-production-build.yml:17]: nice.

File: tests/fixtures/phpunit/testmodule/modules/testsubmodule/tests/src/Functional/UserCreateTest.php [hunk 5]
```diff
@@ -37,9 +37,10 @@ class UserCreateTest extends BrowserTestBase {
     public function testUserAdd() {
         $user = $this->drupalCreateUser(['administer users']);
         $this->drupalLogin($user);
+        $request_time = \Drupal::time()->getRequestTime();
 
-        $this->assertEquals(REQUEST_TIME, $user->getCreatedTime(), 'Creating a user sets default "created" timestamp.');
-        $this->assertEquals(REQUEST_TIME, $user->getChangedTime(), 'Creating a user sets default "changed" timestamp.');
+        $this->assertEquals($request_time, $user->getCreatedTime(), 'Creating a user sets default "created" timestamp.');
+        $this->assertEquals($request_time, $user->getChangedTime(), 'Creating a user sets default "changed" timestamp.');
```
Comment (deviantintegral) [inline tests/fixtures/phpunit/testmodule/modules/testsubmodule/tests/src/Functional/UserCreateTest.php:43]: 👏

**Review by deviantintegral — APPROVED** [review]
I wonder if we should consider this a BC break, and bump Drainpipe to v4? In particular, if the twig linter has any sort of different rules, or if teams were expecting the twig linter in other custom tasks.

File: .github/workflows/TestMetapackages.yml (hunk 3 above)
Comment (justafish) [reply .github/workflows/TestMetapackages.yml:280]: It's testing whether the package import works correctly when bundling

**Comment by justafish** [comment]
@deviantintegral I've made some tweaks to make it compatible with both D10 and D11. I don't think we need to bump to v4 as the twig linter wasn't configurable previously.

**Pull request merged**